test/lib/mayaUsd/fileio/testEditAsMaya.py

```diff
@@ -135,5 +135,38 @@ def testIllegalEditAsMaya(self):
         # self.assertFalse(mayaUsd.lib.PrimUpdaterManager.canEditAsMaya(scopePathStr))
         # self.assertFalse(mayaUsd.lib.PrimUpdaterManager.editAsMaya(scopePathStr))
 
+    @unittest.skipIf(os.getenv('UFE_PREVIEW_VERSION_NUM', '0000') < '3006', 'Test only available in UFE preview version 0.3.6 and greater')
+    def testSessionLayer(self):
+        '''Verify that the edit gets on the sessionLayer instead of the editTarget layer.'''
+        
+        import mayaUsd_createStageWithNewLayer
+
+        proxyShapePathStr = mayaUsd_createStageWithNewLayer.createStageWithNewLayer()
+        stage = mayaUsd.lib.GetPrim(proxyShapePathStr).GetStage()
+        sessionLayer = stage.GetSessionLayer()
+        prim = stage.DefinePrim('/A', 'Xform')
+
+        primPathStr = proxyShapePathStr + ',/A'
+
+        self.assertTrue(stage.GetSessionLayer().empty)
+
+        self.assertTrue(mayaUsd.lib.PrimUpdaterManager.canEditAsMaya(primPathStr))
+        self.assertTrue(mayaUsd.lib.PrimUpdaterManager.editAsMaya(primPathStr))
+
+        self.assertFalse(stage.GetSessionLayer().empty)
+
+        kPullPrimMetadataKey = "Maya:Pull:DagPath"
+        with Usd.EditContext(stage, stage.GetSessionLayer()):
+            self.assertEqual(prim.GetCustomDataByKey(kPullPrimMetadataKey), "|__mayaUsd__|AParent|A")
+
+        # Discard Maya edits, but there is nothing to discard.
+        self.assertTrue(mayaUsd.lib.PrimUpdaterManager.discardEdits("A"))
+
+        # Now Session Layer should be empty, but it is not, it has an empty content
+        self.assertFalse(stage.GetSessionLayer().empty)
```

Comment:
What is this empty content?

Reply:
// USD Layer identifier: anon:00000217FCAC6AE0:anonymousLayer1-session.usda
// Real Path: 
// #usda 1.0
// 
// over "A"
// {
// }
//

Reply:
O.K., this is what I suspected.  We should be doing this:
https://graphics.pixar.com/usd/release/api/class_sdf_layer.html#a55468b05043d021961a36fdf4912863d
to remove the inert primSpec.  I feel this is out of scope for your pull request, so I'll enter a separate JIRA ticket for it

Reply:
Entered as MAYA-120677.